### DISCUSSION TITLE
on_member_join event not triggering

### DISCUSSION THREAD

**Discussion by Zaymus** [Rapptz/discord.py#6125] [Q&A] [answered]
I'm trying to make my discord bot welcome people that join a server however I'm having issues getting the event to trigger.

    import discord
    import os

    intents = discord.Intents.all()

    f = open("C:\\Users\\carso\\Desktop\\discord_token.txt", "r")
    token = f.read()

    class MyClient(discord.Client):
        main_channel = ''

        async def on_ready(self):
            main_channel = self.get_channel(520032652401704960)
            print(f'{self.user} is now active')

        async def on_member_join(self, member):
            print("user joined")
            await main_channel.send(f'{member.username} has joined the server')
            await self.send_message(member, "welcome")

    client = MyClient()
    client.run(token)#runs the bot using the token that was read from the txt file

**Comment by Harmon758** [comment]
Have you enabled the server members intent in your bot's dashboard?

**Reply by Zaymus** [reply]
Yes I have

**Reply by Nemika-Haj** [reply]
You have forgotten the __init__ class. You obviously need to define the intents in the code as well. Try something like;
```py
def __init__(self):
    self.intents = intents
```

**Comment by Harmon758** [comment, marked as answer]
You need to pass the intents when you initialize [`Client`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Client).

**Reply by Zaymus** [reply]
wow i'm dumb lol, I cant believe I missed that thanks for the help
